src/tiled/mapscene.cpp

```diff
@@ -305,7 +306,11 @@ void MapScene::updateSceneRect()
                      margins.bottom());
 
     setSceneRect(sceneRect);
-    mDarkRectangle->setRect(sceneRect);
+
+    if (mMapDocument->map()->infinite())
+        mDarkRectangle->setRect(QRect());
```

Comment:
We'll need to find some way to still darken the view, for the "Highlight Current Layer" feature to work on infinite maps.